### ISSUE TITLE
NativeMethods should use libdl.so.2 instead of libdl.so

### ISSUE THREAD

**Issue by lyonsil** [sillsdev/icu-dotnet#185]
## Describe the bug

ICU.net fails to initialize on Ubuntu version >= 22.  The exception says "System.DllNotFoundException: Unable to load shared library 'libdl.so' or one of its dependencies".  Here is a stack trace:
```
   at Icu.NativeMethods.dlopen(String file, Int32 mode)
   at Icu.NativeMethods.GetIcuLibHandle(String basename, Int32 icuVersion)
   at Icu.NativeMethods.LoadIcuLibrary(String libraryName)
   at Icu.NativeMethods.get_IcuCommonLibHandle()
   at Icu.NativeMethods.u_init(ErrorCode& errorCode)
   at Icu.Wrapper.Init()
```

## To Reproduce

Call Init() on Ubutunu.

## Expected behavior

ICU.net should use libdl.so.2.  This issue has been seen by others in different projects on SO: https://stackoverflow.com/questions/75855053/how-to-address-crash-due-to-missing-libdl-so-on-ubuntu-22.

**Comment by lyonsil** [comment]
I was using version 2.8.1, and a fix for this was already in 2.9.0.

**Comment by ermshiperete** [comment]
Note that the mapping from `libdl.so` to `libdl.so.2` is done in the `icu.net.dll.config` which has to exist next to `icu.net.dll` when running on Mono and .NET Core 3.x/.NET 5+.
